src/Inventory/Projectile.cpp

```diff
@@ -0,0 +1,81 @@
+//
+// Created by musique88 on 12/30/20.
+//
+
+#include <Utils/Log.h>
+#include <glm/gtx/string_cast.hpp>
+#include "Inventory/Projectile.h"
+
+namespace CC::Entities
+{
+	Projectile::Projectile(
+			uint texMeshId, Element element, bool explodes, uint duration, float damage, float size,
+			glm::vec3 position, glm::vec3 rotation,glm::vec3 velocity)
+		:CGE::Entities::Entity(texMeshId, position, rotation),
+		velocity_(velocity), element_(element), explodes_(explodes), duration_(duration), damage_(damage), size_(size)
+	{
+		setSpeed(velocity);
```

Comment:
Why you set speed after setting the velocity? SetSpeed => as_ = speed, so just put as_(velocity) in you initializer list.